backend/modules/publishing/router.py

```
@@ -261,127 +332,65 @@ async def build_simulation_response(simulation: Simulation, db: Session) -> Dict
-                    f"in S3 (checked jpg, png, webp)"
-                )
+            scene_id, s3_url = result
+            if s3_url and scene_id in scene_map:
+                scene_map[scene_id].image_url = s3_url
+                db.add(scene_map[scene_id])
+                updates_made = True
+        
+        # Single commit for all updates (instead of N commits)
+        if updates_made:
+            db.commit()
+            logger.info(f"[S3_PARALLEL] ✅ Committed all S3 URL updates")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)